ngx_http_sxg_filter_module.c

```diff
@@ -537,15 +556,20 @@ static ngx_int_t ngx_http_sxg_header_filter(ngx_http_request_t* req) {
     if (ctx == NULL) {
       return NGX_ERROR;
     }
+
+    if (pthread_mutex_init(&ctx->lock, NULL) != 0) {
```

Comment:
Should this call `pthread_mutex_destroy` somewhere?

Reply:
OK, we should destroy at the end of every requests.